### PR TITLE
Basic M1M3 TS commands

### PR DESCRIPTION
**Pull request by pkubanek**

### CODE REVIEW

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Telemetry.xml [hunk 1]
```diff
@@ -12,32 +13,53 @@
...
-      <Description>Thermocouple scanners 2 read values.</Description>
+      <EFDB_Name>heaterDisabled</EFDB_Name>
+      <Description>Heater disabled.</Description>
+      <IDL_Type>boolean</IDL_Type>
+      <Units>unitless</Units>
+      <Count>96</Count>
+    </item>
+    <item>
+      <EFDB_Name>heaterBreaker</EFDB_Name>
+      <Description>Heater breaker (false=closed, true=open).</Description>
```
[the remark this answers is not in the document]
Comment (r-owen) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Telemetry.xml:31]: Similarly for fanBreaker

**Review by r-owen — CHANGES_REQUESTED** [review]
Please rebase to develop and remove deprecated fields.
I also hope you can clarify some of the descriptions.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Commands.xml [hunk 2]
```diff
@@ -14,10 +14,104 @@
   </SALCommand>
   <SALCommand>
     <Subsystem>MTM1M3TS</Subsystem>
-    <EFDB_Topic>MTM1M3TS_command_shutdown</EFDB_Topic>
+    <EFDB_Topic>MTM1M3TS_command_enterEngineering</EFDB_Topic>
+    <Alias>enterEngineering</Alias>
+    <Explanation>http://sal.lsst.org</Explanation>
```
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Commands.xml]: Removed

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Commands.xml [hunk 3]
```diff
@@ -14,10 +14,104 @@
...
+  </SALCommand>
+  <SALCommand>
+    <Subsystem>MTM1M3TS</Subsystem>
+    <EFDB_Topic>MTM1M3TS_command_heaterFanDemand</EFDB_Topic>
+    <Alias>heaterFanDemand</Alias>
+    <Explanation>http://sal.lsst.org</Explanation>
+    <Description>Sets thermal unit heater PWM and fan RPM demand values. Command code 88 (0x58).</Description>
+    <item>
+      <EFDB_Name>heaterPWM</EFDB_Name>
+      <Description>Heater PWM demand offset. 0-255, where 255 equals 100%.</Description>
```
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Commands.xml:45]: You are correct. Offset remained there from the time when it wasn't clear how PWM and RPM would be demanded.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml [hunk 4]
```diff
@@ -15,11 +15,452 @@
...
   <SALEvent>
     <Subsystem>MTM1M3TS</Subsystem>
-    <EFDB_Topic>MTM1M3TS_logevent_detailedState</EFDB_Topic>
+    <EFDB_Topic>MTM1M3TS_logevent_reHeaterGains</EFDB_Topic>
     <item>
-      <EFDB_Name>detailedState</EFDB_Name>
-      <Description>Detailed state of the CSC</Description>
-      <IDL_Type>long</IDL_Type>
+      <EFDB_Name>proportionalGain</EFDB_Name>
+      <Description>Re-Heater proportional gain.</Description>
```
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml:35]: Done. Forgot about timestamps, as I still consider those automatic - added them

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Commands.xml [hunk 5]
```diff
@@ -14,10 +14,104 @@
   </SALCommand>
   <SALCommand>
     <Subsystem>MTM1M3TS</Subsystem>
-    <EFDB_Topic>MTM1M3TS_command_shutdown</EFDB_Topic>
+    <EFDB_Topic>MTM1M3TS_command_enterEngineering</EFDB_Topic>
+    <Alias>enterEngineering</Alias>
+    <Explanation>http://sal.lsst.org</Explanation>
+    <Description>Enters engineering detailedState. In this states direct ILC controll is allowed.</Description>
```
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Commands.xml]: I removed enter/exitEngineering. They are currently not needed.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml [hunk 6]
```diff
@@ -15,11 +15,452 @@
...
+    <item>
+      <EFDB_Name>timestamp</EFDB_Name>
+      <Description>Thermal warning timestamp</Description>
+      <IDL_Type>double</IDL_Type>
+      <Units>unitless</Units>
+      <Count>1</Count>
+    </item>
+    <item>
+      <EFDB_Name>anyWarning</EFDB_Name>
+      <Description>Is there a warning on the thermal fan unit?</Description>
```
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml:113]: Changed to:

```
      <Description>Does at least a single thermal fan unit indicates any warning?</Description>
```

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml [hunk 7]
```diff
@@ -15,11 +15,452 @@
...
+  <SALEvent>
+    <Subsystem>MTM1M3TS</Subsystem>
+    <EFDB_Topic>MTM1M3TS_logevent_thermalInfo</EFDB_Topic>
+    <item>
+      <EFDB_Name>timestamp</EFDB_Name>
+      <Description>Thermal Info timestamp</Description>
+      <IDL_Type>double</IDL_Type>
+      <Units>unitless</Units>
+      <Count>1</Count>
+    </item>
```
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml]: I removed all timestamps from events.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml [hunk 8]
```diff
@@ -15,11 +15,452 @@
...
+    <item>
+      <EFDB_Name>ilcUniqueId</EFDB_Name>
+      <Description>Thermal ILC unique id</Description>
+      <IDL_Type>long long</IDL_Type>
+      <Units>unitless</Units>
+      <Count>96</Count>
+    </item>
+    <item>
+      <EFDB_Name>ilcApplicationType</EFDB_Name>
+      <Description>Thermal ILC application type</Description>
```
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml]: The field is described in LTS-646, it's different for TS, FA and HP (Thermal, Force Actuators and HardPoints) ILCs. It's almost enumeration, but this is really just for completenes - having anything else then TS application loaded in TS ILCs will cause a disaster. My preference is to leave as it is.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Telemetry.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Telemetry.xml]: Thanks. I leave open/closed, as this makes more sence for eletrician, and added OK/tripped:

```xml
      <Description>Heater circuit breaker (false=closed/OK, true=open/tripped).</Description>
      <IDL_Type>boolean</IDL_Type>
      <Units>unitless</Units>
      <Count>96</Count>
    </item>
    <item>
      <EFDB_Name>fanBreaker</EFDB_Name>
      <Description>Fan circuit breaker (false=closed/OK, true=open/tripped).</Description>
```

**Comment by rbovill** [comment]
@pkubanek Could you squash the commits into one?

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml (hunk 8 above)
[the remark this answers is not in the document]
Comment (r-owen) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml]: Could you please at least add a reference to LTS-646? Though I am unhappy at having to look up a document just to figure out what this field means (and I'm not going to go look). The XML should be as self-contained as practical.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Telemetry.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (r-owen) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Telemetry.xml]: That's great. I didn't realize electricians used closed/open for circuit breakers.

**Review by r-owen — CHANGES_REQUESTED** [review]
Thank you for the many improvements. There are still a few items I hope you will change.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml (hunk 8 above)
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml]: I don't know about assigning values. How can that be done? Any example?

[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml]: Added enumeration. Closing this comment. Thanks.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Commands.xml [hunk 9]
```diff
@@ -14,10 +14,104 @@
...
+    </item>
+  </SALCommand>
+  <SALCommand>
+    <Subsystem>MTM1M3TS</Subsystem>
+    <EFDB_Topic>MTM1M3TS_command_turnPowerOn</EFDB_Topic>
+    <Alias>turnPowerOn</Alias>
+    <Explanation>http://sal.lsst.org</Explanation>
+    <Description>Turn power on for fail/heaters coils and coolant pump. Controls cRIO slot 5 SSR.</Description>
+    <item>
+      <EFDB_Name>turnPowerFanCoilsHeatersOn</EFDB_Name>
```
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Commands.xml]: Renamed as suggested, hopefully solved.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <?xml-stylesheet type="text/xsl" href="http://lsst-sal.tuc.noao.edu/schema/SALEventSet.xsl"?>
 <SALEventSet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://lsst-sal.tuc.noao.edu/schema/SALEventSet.xsd">
-  <Enumeration>DetailedStates_DisabledState,DetailedStates_EnabledState,DetailedStates_FaultState,DetailedStates_OfflineState,DetailedStates_StandbyState</Enumeration>
+  <Enumeration>SummaryStates_DisabledState,SummaryStates_EnabledState,SummaryStates_FaultState,SummaryStates_OfflineState,SummaryStates_StandbyState</Enumeration>
```
[the remark this answers is not in the document]
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml]: Removed.

**Comment by pkubanek** [comment]
> @pkubanek Could you squash the commits into one?

git rebase -i develop, changed pick to s and git push -f.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml [hunk 11]
```diff
@@ -1,7 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <?xml-stylesheet type="text/xsl" href="http://lsst-sal.tuc.noao.edu/schema/SALEventSet.xsl"?>
 <SALEventSet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://lsst-sal.tuc.noao.edu/schema/SALEventSet.xsd">
-  <Enumeration>DetailedStates_DisabledState,DetailedStates_EnabledState,DetailedStates_FaultState,DetailedStates_OfflineState,DetailedStates_StandbyState</Enumeration>
+  <Enumeration>
+    ILCType_ElectromechanicalActuator=1,
+    ILCType_PneumaticActuator=2,
+    ILCType_Thermal=3,
+    ILCType_HardpointMonitoringa=7,
```
Comment (r-owen) [inline sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml:8]: Thank you for these enums.

Is the "a" at the end of `HardpointMonitoringa` a typo?

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml [hunk 12]
```diff
@@ -15,11 +21,429 @@
...
+    <item>
+      <EFDB_Name>ilcApplicationType</EFDB_Name>
+      <Description>Thermal ILC application type. ILCType enumeration.</Description>
+      <IDL_Type>short</IDL_Type>
+      <Units>unitless</Units>
+      <Count>96</Count>
+    </item>
+    <item>
+      <EFDB_Name>networkNodeType</EFDB_Name>
+      <Description>Thermal ILC network node type. ILCType enumeration. Mismatch with ilcApplicationType indicates ILC programming/seting problem.</Description>
```
Comment (r-owen) [inline sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml:94]: Nice and clear!

**Review by r-owen — APPROVED** [review]
This looks great.

I may have spotted a typo in one enum name (noted in an inline comment). Please have a look before merging.

File: sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml (hunk 11 above)
Comment (pkubanek) [reply sal_interfaces/MTM1M3TS/MTM1M3TS_Events.xml]: Yes, coming from vim commands.

**Pull request merged**